### PR TITLE
13330 Update the Legal API to apply BEN IA correction changes 

### PR DESCRIPTION
**Pull request by lewischenstudio**
*Issue #:* /bcgov/entity#13330

*Description of changes:*
- updated schemas version in requirements.txt
- updated lega-api version

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

**Comment by argush3** [comment]
The report updates were actually intended to be done in #13504 which has a pretty large estimate attached to it.  If you intend to do this work, you should do it in #13504.  We should have taken some of todos in this(filer) ticket that already had separate tickets created.

In general, I think we want to move BEN corrections to use the same approach as what we are doing for SP/GPs.  There is some existing in the reporting logic that needs to be updated to follow model.  I've included some things that I have found from an initial look.  @lekshmimallika-aot @vysakh-menon-aot please chime in if I'm saying anything incorrect or if you have any specific thoughts.

I don't think the report key needs to be changed to contain the value of `correctedFilingType` anymore for a BEN anymore.
https://github.com/bcgov/lear/blob/9894a0929d2c191713fd361417ec843b7b1de30e/legal-api/src/legal_api/reports/report.py#L59-L64


`self._format_with_diff_data(filing)` is no longer required and can follow something more similar to what corrections for registrations is doing.  Perhaps update `_format_alteration_date` function...
https://github.com/bcgov/lear/blob/9894a0929d2c191713fd361417ec843b7b1de30e/legal-api/src/legal_api/reports/report.py#L233-L238

determine if changes need to happen here
https://github.com/bcgov/lear/blob/9894a0929d2c191713fd361417ec843b7b1de30e/legal-api/src/legal_api/reports/report.py#L883-L892

Is there a need to have a new output like how SP/GP has the correct registration statement?
https://github.com/bcgov/lear/blob/5712ddcaf899fb76554a9d4d193050b5de471cc2/legal-api/src/legal_api/core/meta/filing.py#L163-L166

**Comment by lewischenstudio** [comment]
Thanks Argus. I have dropped the reports from this pull request since it will be handled in a larger ticket #13504. For this PR, I only updated the schemas version in the legal api.

**Comment by argush3** [comment]
> The test failure is due to the recent change from `draft7_format_checker` to `Draft7Validator.FORMAT_CHECKER`. Maybe we should use `Draft7Validator.format_checker` since `AttributeError: type object 'Draft7Validator' has no attribute 'FORMAT_CHECKER'` and `format_checker` is a valid attribute?

The business schema was updated to use `jsonschema` version 4.16.0  - https://github.com/bcgov/business-schemas/pull/110.  

Try updating corresponding `jsonschema` entry in requirements.txt to use 4.16.0

https://github.com/bcgov/lear/blob/21f1a52a1fe03984a8e929523921cd9641813a38/legal-api/requirements.txt#L37

File: legal-api/src/legal_api/version.py [hunk 1]
```diff
@@ -22,4 +22,4 @@
 Development release segment: .devN
 """
 
-__version__ = '2.58.0'  # pylint: disable=invalid-name
+__version__ = '2.58.1'  # pylint: disable=invalid-name
```
Comment (argush3) [inline legal-api/src/legal_api/version.py:25]: we usually only bump version numbers before a release.  please revert.

Comment (lewischenstudio) [reply legal-api/src/legal_api/version.py]: Okay. It's reverted.

**Review by argush3 — APPROVED** [review]

**Pull request merged**